File: testdata/precision-suite/tp_alias_idioms.cs
// True positives for the C# `using X = Y;` alias.
//
// C# sinks are named by TYPE in the catalog (`Process.Start`), which is what
// code writes after a plain `using System.Diagnostics;`. A using-ALIAS renames
// the type and made the same call invisible.
//
// Known limit, deliberately not papered over: an alias that names a NAMESPACE
// (`using D = System.Diagnostics;` then `D.Process.Start`) resolves to neither
// the bare nor the fully-qualified form the catalog carries, and is still
// missed. Only the type-alias case below is claimed.
using Proc = System.Diagnostics.Process;

class AliasIdioms
{
    void AliasedStart(HttpRequest Request)
    {
        var user = Request.QueryString["cmd"];
        Proc.Start("sh", "-c " + user);  // nox-expect: TAINT-002
    }
}

File: testdata/precision-suite/tp_alias_idioms.ex
# True positives for the Elixir `alias …, as:` idiom.
#
# Elixir sinks are named BARE in the catalog (`System.cmd`, `Repo.query`),
# which is what unaliased code writes. An `as:` rename made the same call
# invisible, so this sample resolves to the last segment rather than the whole
# path -- the form the catalog actually carries.
#
# The suite had no Elixir sample, so nothing measured this either way.
defmodule Precision.AliasIdioms do
  alias System, as: Sys

  def aliased_cmd(conn) do
    cmd = conn.params
    Sys.cmd("sh", ["-c", cmd])  # nox-expect: TAINT-002
  end
end
